### PR TITLE
Emit dns servers in DHCPv4 repr. Fixes #504

### LINKED ISSUE #504: DHCPRepr doesn't include DNS servers when emitting

### PR DESCRIPTION
**Pull request by theli-ua**

### CODE REVIEW

**Comment by theli-ua** [comment]
ok, no alloc, right, let me fix that

File: src/wire/dhcpv4.rs
Comment (Dirbaio) [inline src/wire/dhcpv4.rs]: simpler to do `.iter().flatten().count() * 4`

File: src/wire/dhcpv4.rs
Comment (Dirbaio) [inline src/wire/dhcpv4.rs]: It's nice to avoid `unwrap`s if you can. You can get rid of this one if you do `dns_servers.iter().flatten()`

File: src/wire/dhcpv4.rs
Comment (Dirbaio) [inline src/wire/dhcpv4.rs]: returning `i+1` to then do `.last()` is a bit complex. You can return `()` and then do `.count()`.

**Review by Dirbaio — CHANGES_REQUESTED** [review]

**Comment by theli-ua** [comment]
@Dirbaio I have a question re source formatting, it is a bit of a pain that one can't just do rustfmt. Is that intended?

**Comment by theli-ua** [comment]
all clippy warnings are from code thats already there

**Comment by crawford** [comment]
@theli-ua do you mind fixing the clippy warnings while you are in here? Regarding `rustfmt`, the maintainers have a preferred style that isn't easily expressible and have opted to format the code by hand.

File: src/wire/dhcpv4.rs
Comment (crawford) [inline src/wire/dhcpv4.rs]: I believe you need to use inspect() instead.

**Comment by theli-ua** [comment]
manual formatting seem to serves as unnecessary friction and deterrent for potential contributions  tbh.

I fixed clippy warnings, but really they were all over the place and have nothing to do with the change in this pr

**Comment by crawford** [comment]
Hmm, that was a much larger fix than I expected. I only saw a single clippy error from https://github.com/smoltcp-rs/smoltcp/pull/505/commits/e023164908681e6ebce2a229bb5d7bca5e4ec735 (though, I realize now you mentioned warnings, not errors). I think you can probably drop that last commit from this PR.

**Comment by theli-ua** [comment]
@crawford I've removed last commit and all checks are passing. Let me know if there is more feedback

**Comment by crawford** [comment]
Overall, this seems fine to me (might want to squash the commit history in this PR though), but I’m not a maintainer so I must defer to @Dirbaio.

**Review by Dirbaio — APPROVED** [review]
Looks great, thank you!

**Comment by theli-ua** [comment]
@Dirbaio what's the current approach for 0.7.x maintenance releases?

**Comment by Dirbaio** [comment]
If you want, send a PR against the `v0.7.x` branch cherry-picking this change, and I'll release it.

**Comment by theli-ua** [comment]
@Dirbaio 0.7.x *is* the branch it was merged into :D

**Comment by Dirbaio** [comment]
Oh! :rofl: 

Okay, let me release and cherry-pick into master.

**Comment by Dirbaio** [comment]
v0.7.5 now out

**Comment by Dirbaio** [comment]
@theli-ua next time send PRs to master, we later cherry-pick cherrypickable features back to `v0.7.x`, not the other way around.

**Pull request merged**
